src/systems/systems/src/hydraulic/brake_circuit.rs

```diff
@@ -0,0 +1,774 @@
+                let remaining_vol_after_accumulator_empty = delta_vol - volume_from_acc;
+                self.volume_to_actuator_accumulator += remaining_vol_after_accumulator_empty;
+            }
+        } else {
+            // Else case if no accumulator: we just take deltavol needed or return it back to res
+            self.volume_to_actuator_accumulator += delta_vol;
+        }
+
+        self.volume_to_res_accumulator += self.left_brake_actuator.reservoir_return();
+        self.volume_to_res_accumulator += self.right_brake_actuator.reservoir_return();
+
+        self.left_brake_actuator.reset_accumulators();
+        self.right_brake_actuator.reset_accumulators();
+
+        self.pressure_applied_left = self.left_brake_actuator.get_applied_brake_pressure();
+        self.pressure_applied_right = self.right_brake_actuator.get_applied_brake_pressure();
+
+        self.accumulator_fluid_pressure_sensor_filtered = self
+            .accumulator_fluid_pressure_sensor_filtered
+            + (actual_pressure_available - self.accumulator_fluid_pressure_sensor_filtered)
+                * (1.
+                    - E.powf(
+                        -context.delta_as_secs_f64()
+                            / BrakeCircuit::ACC_PRESSURE_SENSOR_FILTER_TIMECONST,
+                    ));
+    }
+
+    pub fn set_brake_demand_left(&mut self, brake_ratio: f64) {
+        self.demanded_brake_position_left = brake_ratio.min(1.0).max(0.0);
+    }
+
+    pub fn set_brake_demand_right(&mut self, brake_ratio: f64) {
+        self.demanded_brake_position_right = brake_ratio.min(1.0).max(0.0);
+    }
+
+    pub fn left_brake_pressure(&self) -> Pressure {
+        self.pressure_applied_left
+    }
+
+    pub fn right_brake_pressure(&self) -> Pressure {
+        self.pressure_applied_right
+    }
+
+    fn accumulator_pressure(&self) -> Pressure {
+        self.accumulator_fluid_pressure_sensor_filtered
+    }
+
+    pub fn accumulator_fluid_volume(&self) -> Volume {
+        self.accumulator.fluid_volume()
+    }
+
+    pub fn reset_accumulators(&mut self) {
+        self.volume_to_res_accumulator = Volume::new::<gallon>(0.);
+        self.volume_to_actuator_accumulator = Volume::new::<gallon>(0.);
+    }
+}
+impl Actuator for BrakeCircuit {
+    fn used_volume(&self) -> Volume {
+        self.volume_to_actuator_accumulator
+    }
+    fn reservoir_return(&self) -> Volume {
+        self.volume_to_res_accumulator
+    }
+}
+impl SimulationElement for BrakeCircuit {
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write_f64(&self.id_left_press, self.left_brake_pressure().get::<psi>());
+        writer.write_f64(
+            &self.id_right_press,
+            self.right_brake_pressure().get::<psi>(),
+        );
+        if self.has_accumulator {
+            writer.write_f64(&self.id_acc_press, self.accumulator_pressure().get::<psi>());
+        }
+    }
+}
+
+#[cfg(test)]
+struct AutoBrakeController {
```

Comment:
**issue**: `AutoBrakeController` is unused outside of tests. I've flagged it with `#[cfg(test)]`. Note that `num_of_modes` remains unused, and this will have to be resolved in any case.
**suggestion**: Remove the `AutoBrakeController` and add it back when creating an auto brake related PR. That way, the reviewer at that time will have a full picture while doing the review.

Reply:
Will try to branch this on my side! Hope we find a solution to make our brakes brake in the sim sooner than later.